18-websockets/classworks/Gavruk_Alexander/server/index.js

```diff
@@ -0,0 +1,31 @@
+const app = require('express')();
+const http = require('http').createServer(app);
+const io = require('socket.io')(http);
+
+io.on('connection', (socket) => {
+  console.log('New client connected');
+
+  socket.emit('username', { username: 'Anonymous' });
+
+  socket.on('set_username', (username) => {
+    socket.username = username;
+    socket.emit('username', { username: socket.username });
+  });
+
+  socket.on('new_message', (msg) => {
+    socket.lastMessage = msg;
+    io.sockets.emit('messages', socket.lastMessage);
```

Comment:
I think it's not necessary to send 'message' event to everybody. You can send it to all users instead of current.